src/dfcx_scrapi/core/agents.py

```diff
@@ -358,6 +366,11 @@ def export_agent(self, agent_id: str, gcs_bucket_uri: str) -> str:
         request = types.agent.ExportAgentRequest()
         request.name = agent_id
         request.agent_uri = gcs_bucket_uri
+        if environment:
+            self._environments = environments.Environments(creds=self.creds)
```

Comment:
@kmaphoenix are we trying to move away from passing credentials explicitly? If so, should this be changed to not pass anything to Environments?

Reply:
@Omerside internally in each file, the creds _should_ be passed explicitly to maintain the state of the original Auth token. So this should remain as-is.

You're likely referring to not passing creds when Instantiating classes as an ad hoc script.  
In those scenarios, we want to enable the scripts to resolve Auth on their own by finding the authorized user via env vars instead of explicitly passing them.

So there's a nuance to understand, but I think it is a critical one.